src/controller/filter.ts

```diff
@@ -0,0 +1,79 @@
+import { FilterModel, FilterModelProperty } from "../models/filter/filter-model";
+import { FilterConfig } from "../models/filter/filter-config-item";
+import {FilterView} from "../views/interfaces";
+import {EventManager} from "../event-manager/event-manager";
+
+export class Filter<T> {
```

Comment:
User shouldn't be able to set min value greater than max value. Please add validation for this test case. 

![image](https://user-images.githubusercontent.com/25628390/71346917-28dc8580-257a-11ea-867e-ac382c20a48c.png)